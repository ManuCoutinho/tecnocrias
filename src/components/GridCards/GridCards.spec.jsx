import '@testing-library/jest-dom'
import { screen, render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import { Cards } from '.'

const propsMock = [
  {
    title: 'title test',
    description: 'some description'
  }
]

const allPropsMock = [
  {
    title: 'title test',
    description: 'some description',
    href: '#anchor',
    link: 'the moon'
  }
]
describe('<GridCards/>', () => {
  it('should render GridCards correctly  with basic props', () => {
    render(<Cards cardList={propsMock} />)
    expect(screen.getByTestId('grid-cards')).toBeInTheDocument()
    expect(screen.getByText(/some description/i)).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument()
    expect(screen.queryByRole('link')).not.toBeInTheDocument()
  })
  it('should render GridCards correctly with all props', () => {
    render(<Cards cardList={allPropsMock} />)
    expect(screen.getByRole('link', { name: /moon/i })).toHaveAttribute(
      'href',
      '#anchor'
    )
  })
  it('should match to snapshot', () => {
    const { container } = render(<Cards cardList={allPropsMock} />)
    expect(container).toMatchSnapshot()
  })
})
