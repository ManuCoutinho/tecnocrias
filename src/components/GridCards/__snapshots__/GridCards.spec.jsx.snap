// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridCards/> > should match to snapshot 1`] = `
<div>
  <div
    class="grid"
    data-testid="grid-cards"
  >
    <div
      class="grid__card"
    >
      <h4
        class="grid__title"
      >
        title test
      </h4>
      <p
        class="grid__description"
      >
        some description
         
        <a
          href="#anchor"
          rel="noopener noreferrer"
        >
          the moon
        </a>
      </p>
    </div>
  </div>
</div>
`;
